### PR TITLE
Stop using the `result` field in `finished.json`

### PR DESCRIPTION
**Pull request by stevekuznetsov**
We always have the `passed` boolean in `finished.json` and should stop
using the Jenkins-style `result` string wherever possible as it contains
the same information and is otherwise useless.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @BenTheElder @cjwagner 
/assign @rmmh 
/area gubernator

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
We should work on the Gubernator test output .... it's a bit of a mess. >.<

**Comment by stevekuznetsov** [comment]
@rmmh can you give insight into that failing unit test?

```py
    def test_build_optional_log(self):
        write(self.BUILD_DIR + 'build-log.txt', 'error or timeout or something')
        response = self.get_build_page()
        self.assertIn('<a href="?log#log">', response)
        self.assertNotIn('timeout', response)
        self.assertNotIn('build-log.txt', response)
```

```
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/workspace/k8s.io/test-infra/gubernator/view_build_test.py", line 231, in test_build_optional_log
    self.assertNotIn('timeout', response)
  File "/usr/lib/python2.7/unittest/case.py", line 810, in assertNotIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 410, in fail
    raise self.failureException(msg)
AssertionError: 'timeout' unexpectedly found in <200 OK text/html body='<!DOCTYPE...v>\n'/3314>
```

File: gubernator/OWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+approvers:
+- rmmh
+- spxtr
```
Comment (rmmh) [inline gubernator/OWNERS]: probably not spxtr

File: gubernator/test-gubernator.sh [hunk 2]
```diff
@@ -20,7 +20,8 @@ set -o xtrace
 
 cd "$(dirname "$0")"
 pip install -r test_requirements.txt
-./test.sh --nologcapture
+mkdir -p "${WORKSPACE}/_artifacts"
+./test.sh --nologcapture --with-xunit --xunit-file="${WORKSPACE}/_artifacts/junit_nosetests.xml"
```
Comment (rmmh) [inline gubernator/test-gubernator.sh]: cute!

**Comment by rmmh** [comment]
I'm not sure that *all* of our jobs have this-- this is a breaking API change, and Gubernator displays build results for some projects that don't have passed fields at all.

**Comment by rmmh** [comment]
The failing unit tests verifies that a passing build doesn't parse the build log.

File: gubernator/OWNERS (hunk 1 above)
Comment (stevekuznetsov) [reply gubernator/OWNERS]: Sure, just grabbed top committers.

**Comment by stevekuznetsov** [comment]
I'll add in a fake on load so we can be sure `passed` boolean is present regardless of what we load. I don't really understand that unit test -- can you please explain what it is doing? If there is a build log it should not be shown on the page?

**Comment by rmmh** [comment]
Grep for SUCCESS or FAILURE in gubernator/ -- you missed a few uses.

**Comment by stevekuznetsov** [comment]
The tests passing in `{'result': 'SUCCESS'}` that were not failing probably have poor coverage. In any case ... those tests are fine. I can switch them to pass `{'passed': True}`. Can you please explain the intent of the test that is pasted above and failing?

**Comment by rmmh** [comment]
Passing builds should not show build logs by default. The test verifies that.

**Comment by stevekuznetsov** [comment]
Finally! Jeez! Some of those little things snaked out across a lot of methods ...


@rmmh PTAL

File: gubernator/view_build.py [hunk 3]
```diff
@@ -112,6 +112,39 @@ def get_running_build_log(job, build, prow_url):
...
+    and finished.json files from a GCS bucket.
+
+    :param started_future: future from gcs_async.read()
+    :param finished_future: future from gcs_async.read()
+    :return: started, finished dictionaries
+    """
+    started = started_future.get_result()
+    finished = finished_future.get_result()
+    if finished and not started:
+        started = 'null'
```
Comment (rmmh) [inline gubernator/view_build.py]: this still won't work for view_pr.py

File: gubernator/view_build.py [hunk 4]
```diff
@@ -112,6 +112,39 @@ def get_running_build_log(job, build, prow_url):
...
+    if started and not finished:
+        finished = 'null'
+    elif not (started and finished):
```
Comment (rmmh) [inline gubernator/view_build.py]: this is wrong-- you should return partial results (usually, started and not finished means a build is in progress)

File: gubernator/view_build.py [hunk 5]
```diff
@@ -112,6 +112,39 @@ def get_running_build_log(job, build, prow_url):
...
+
+    :param started_future: future from gcs_async.read()
+    :param finished_future: future from gcs_async.read()
+    :return: started, finished dictionaries
+    """
+    started = started_future.get_result()
+    finished = finished_future.get_result()
+    if finished and not started:
+        started = 'null'
+    if started and not finished:
```
Comment (rmmh) [inline gubernator/view_build.py]: supposed to be an elif?

File: gubernator/view_build.py (hunk 3 above)
Comment (stevekuznetsov) [reply gubernator/view_build.py]: Why?

File: gubernator/view_build.py (hunk 5 above)
Comment (stevekuznetsov) [reply gubernator/view_build.py]: Could be, just copied from what existed beforehand and wanted to make minimal changes.

File: gubernator/view_build.py (hunk 3 above)
Comment (stevekuznetsov) [reply gubernator/view_build.py]: (and how do we write a test to expose that)?

File: gubernator/view_build.py [hunk 6]
```diff
@@ -127,17 +160,10 @@ def build_details(build_dir):
...
-    elif not (started and finished):
-        return
-    started = json.loads(started)
-    finished = json.loads(finished)
+    started, finished = normalize_metadata(
+        gcs_async.read(build_dir + '/started.json'),
+        gcs_async.read(build_dir + '/finished.json')
+    )
 
     junit_paths = [f.filename for f in view_base.gcs_ls_recursive('%s/artifacts' % build_dir)
```
Comment (rmmh) [inline gubernator/view_build.py]: this shouldn't happen if started & finished are None

Comment (stevekuznetsov) [reply gubernator/view_build.py]: ACK

**Comment by stevekuznetsov** [comment]
@rmmh addressed comments

File: gubernator/view_build.py [hunk 7]
```diff
@@ -127,17 +160,13 @@ def build_details(build_dir):
...
-        return
-    started = json.loads(started)
-    finished = json.loads(finished)
+    started, finished = normalize_metadata(
+        gcs_async.read(build_dir + '/started.json'),
+        gcs_async.read(build_dir + '/finished.json')
+    )
+
+    if started is None and finished is None:
+        return started, finished, None
```
Comment (rmmh) [inline gubernator/view_build.py]: weak preference: `return None` > `return None, None, None` > `return started, finished, None`.

File: gubernator/view_pr_test.py [hunk 8]
```diff
@@ -16,16 +16,14 @@
 
 import datetime
 import unittest
+from webapp2_extras import securecookie
```
Comment (rmmh) [inline gubernator/view_pr_test.py]: add a newline before this, to separate it from stdlib imports

**Comment by rmmh** [comment]
/lgtm
/hold

feel free to squash

File: gubernator/view_build.py (hunk 7 above)
Comment (stevekuznetsov) [reply gubernator/view_build.py]: Eh, different return types in one method make my head hurt and complicates the API for callers so I think I'll keep this as-is.

**Comment by stevekuznetsov** [comment]
Updated for import formatting

/hold cancel

**Comment by rmmh** [comment]
upstream gazelle error?
/retest

**Pull request merged**